### PR TITLE
Make the contact fields of ContactInformation model unique coupled wi…

### PR DESCRIPTION
**Pull request by ruddi10**
Make contact fields of ContactInformation model unique coupled with their entity type

### CODE REVIEW

File: models/generics/contact_information.py [hunk 1]
```diff
@@ -63,6 +59,13 @@ class Meta:
         """
 
         verbose_name_plural = 'contact information'
+        constraints=[
+            models.UniqueConstraint(fields=['entity_content_type', 'primary_phone_number'], name='unique primary number'),
+            models.UniqueConstraint(fields=['entity_content_type', 'secondary_phone_number'], name='unique secondary number'),
+            models.UniqueConstraint(fields=['entity_content_type', 'email_address'], name='unique email address'),
+            models.UniqueConstraint(fields=['entity_content_type', 'institute_webmail_address'], name='unique institute webmail address'),
+
```
Comment (dhruvkb) [inline models/generics/contact_information.py:67]: Remove the blank line

**Review by dhruvkb — CHANGES_REQUESTED** [review]

**Comment by ruddi10** [comment]
Removed thanks for telling.

**Comment by pradumangoyal** [comment]
I think the name passed in the `name` parameter should not have spaces in between. Better to replace those with underscores. You can rename the migration file semantically. Always check the PR title before opening a PR

**Comment by ruddi10** [comment]
Made the changes

File: models/generics/contact_information.py [hunk 2]
```diff
@@ -63,6 +59,12 @@ class Meta:
         """
 
         verbose_name_plural = 'contact information'
+        constraints=[
+            models.UniqueConstraint(fields=['entity_content_type', 'primary_phone_number'], name='unique_primary_number'),
+            models.UniqueConstraint(fields=['entity_content_type', 'secondary_phone_number'], name='unique_secondary_number'),
+            models.UniqueConstraint(fields=['entity_content_type', 'email_address'], name='unique_email_address'),
+            models.UniqueConstraint(fields=['entity_content_type', 'institute_webmail_address'], name='unique_institute_webmail_address'),
+     ]
```
Comment (dhruvkb) [inline models/generics/contact_information.py:67]: Closing bracket not aligned with opening line.
```suggestion
        constraints=[
            models.UniqueConstraint(fields=['entity_content_type', 'primary_phone_number'], name='unique_primary_number'),
            models.UniqueConstraint(fields=['entity_content_type', 'secondary_phone_number'], name='unique_secondary_number'),
            models.UniqueConstraint(fields=['entity_content_type', 'email_address'], name='unique_email_address'),
            models.UniqueConstraint(fields=['entity_content_type', 'institute_webmail_address'], name='unique_institute_webmail_address'),
        ]
```

**Review by dhruvkb — CHANGES_REQUESTED** [review]

File: models/generics/contact_information.py (hunk 2 above)
Comment (dhruvkb) [reply models/generics/contact_information.py:67]: Also all these lines seem to be way over the 80 character limit.

**Comment by ruddi10** [comment]
Made the required changes

File: models/generics/contact_information.py [hunk 3]
```diff
@@ -60,10 +60,14 @@ class Meta:
 
         verbose_name_plural = 'contact information'
         constraints=[
-            models.UniqueConstraint(fields=['entity_content_type', 'primary_phone_number'], name='unique_primary_number'),
-            models.UniqueConstraint(fields=['entity_content_type', 'secondary_phone_number'], name='unique_secondary_number'),
-            models.UniqueConstraint(fields=['entity_content_type', 'email_address'], name='unique_email_address'),
-            models.UniqueConstraint(fields=['entity_content_type', 'institute_webmail_address'], name='unique_institute_webmail_address'),
+            models.UniqueConstraint(fields=['entity_content_type', 
+            'primary_phone_number'], name='unique_primary_number'),
```
Comment (dhruvkb) [inline models/generics/contact_information.py:64]: This does look overly long but it's the convention followed across all models in the codebase.
```suggestion
            models.UniqueConstraint(
                fields=[
                    'entity_content_type', 
                    'primary_phone_number',
                ], 
                name='unique_primary_number',
            ),
```

**Review by dhruvkb — CHANGES_REQUESTED** [review]

File: models/generics/contact_information.py (hunk 3 above)
Comment (dhruvkb) [reply models/generics/contact_information.py:64]: Note the trailing commas _everywhere_.

**Review by dhruvkb — APPROVED** [review]
I trust @pradumangoyal and @algomaster99 to look into the specifics. LGTM

**Review by pradumangoyal — APPROVED** [review]

**Pull request merged**